Share to group, receiver rename, share to user sometimes reverts the rename

``mergeShare.feature:82`` is failing intermittently in drone. I noticed a few yesterday. The scenario was enabled recently by PR #34200 after having been skipped for some time. Issue #29016 originally reported the intermittent failure, and the scenario was skipped in PR #29018

An example from last night:

https://drone.owncloud.com/owncloud/core/14522/398
```
  Scenario: Merging shares for recipient when shared from outside with group then user and recipient renames in between                                                     # /drone/src/tests/acceptance/features/apiShareManagement/mergeShare.feature:82
    Given user "user0" has created folder "/merge-test-outside-groups-renamebeforesecondshare"                                                                              # FeatureContext::userCreatesFolder()
    When user "user0" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with group "grp1" using the sharing API                                            # FeatureContext::userSharesFileWithGroupUsingTheSharingApi()
    And user "user1" moves folder "/merge-test-outside-groups-renamebeforesecondshare" to "/merge-test-outside-groups-renamebeforesecondshare-renamed" using the WebDAV API # FeatureContext::userMovesFileUsingTheAPI()
    And user "user0" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with user "user1" using the sharing API                                             # FeatureContext::userSharesFileWithUserUsingTheSharingApi()
    Then as user "user1" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"                # WebDavPropertiesContext::asUserFolderShouldContainAPropertyWithValue()
      Cannot find property "oc:permissions"
      Failed asserting that false is true.
And as "user1" folder "/merge-test-outside-groups-renamebeforesecondshare" should not exist # FeatureContext::asFileOrFolderShouldNotExist()
...

--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiShareManagement/mergeShare.feature:82

176 scenarios (175 passed, 1 failed)
2023 steps (2021 passed, 1 failed, 1 skipped)
18m27.84s (16.31Mb)
runsh: Exit code of main run: 1
Makefile:193: recipe for target 'test-acceptance-api' failed
make: *** [test-acceptance-api] Error 1
```

To Do:
- [x] adjust existing test scenario to stop intermittent fails mergeShare.feature:82 #34242
- [ ] fix the bug
- [ ] adjust the test scenario to test properly/fully again

--- a/tests/acceptance/features/apiShareManagement/mergeShare.feature
+++ b/tests/acceptance/features/apiShareManagement/mergeShare.feature
@@ -78,20 +78,13 @@ Feature: sharing
     And as "user1" folder "/merge-test-inside-twogroups-perms (2)" should not exist
     And as "user1" folder "/merge-test-inside-twogroups-perms (3)" should not exist
 
-  @issue-34235 @skipOnLDAP
   Scenario: Merging shares for recipient when shared from outside with group then user and recipient renames in between
     Given user "user0" has created folder "/merge-test-outside-groups-renamebeforesecondshare"
     When user "user0" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with group "grp1" using the sharing API
     And user "user1" moves folder "/merge-test-outside-groups-renamebeforesecondshare" to "/merge-test-outside-groups-renamebeforesecondshare-renamed" using the WebDAV API
     And user "user0" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with user "user1" using the sharing API
-    # issue-34235 bug is that usually user1 continues to see the share with the correct renamed name,
-    # but sometimes reverts to seeing the share with the original name.
-    # when the bug is fixed, delete the following test step, and uncomment the test steps below it
-    Then as "user1" exactly one of these folders should exist
-      | /merge-test-outside-groups-renamebeforesecondshare         |
-      | /merge-test-outside-groups-renamebeforesecondshare-renamed |
-    #Then as user "user1" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"
-    #And as "user1" folder "/merge-test-outside-groups-renamebeforesecondshare" should not exist
+    Then as user "user1" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"
+    And as "user1" folder "/merge-test-outside-groups-renamebeforesecondshare" should not exist
 
   @skipOnLDAP @user_ldap-issue-274
   Scenario: Merging shares for recipient when shared from outside with user then group and recipient renames in between

--- a/tests/acceptance/features/bootstrap/Sharing.php
+++ b/tests/acceptance/features/bootstrap/Sharing.php
@@ -85,16 +85,14 @@ trait Sharing {
 	 * @return void
 	 */
 	private function waitToCreateShare() {
-		$time = \time();
 		if (($this->localLastShareTime !== null)
-			&& (($time - $this->localLastShareTime) < 1)
+			&& ((\microtime(true) - $this->localLastShareTime) < 1)
 		) {
 			// prevent creating two shares with the same "stime" which is
 			// based on seconds, this affects share merging order and could
 			// affect expected test result order
 			\sleep(1);
 		}
-		$this->localLastShareTime = $time;
 	}
 
 	/**
@@ -666,6 +664,7 @@ trait Sharing {
 			$this->sharingApiVersion
 		);
 		$this->lastShareData = $this->getResponseXml();
+		$this->localLastShareTime = \microtime(true);
 	}
 
 	/**

